src/analyses/base.ml

```diff
@@ -807,6 +809,9 @@ struct
     | TNamed ({ttype=t}, _) -> init_value a gs st t
     | _ -> `Top
 
+  (* We can't just set all to `Bot since some domains/analyses rely on the abstract value having the right type (would make 30 tests fail), so we create a bot of the right abstract type. *)
```

Comment:
Why are we keeping the old `init_value` around? Does not seem to be used anywhere

Reply:
This whole `{bot,top,init}_value` duplication is ugly. Once it's clear what to do, we want some option for whether we assume the program to be free from undefined behavior and set the initial value depending on that.
The question is how `init_value` is different from `top_value`:
~~~ocaml
    | t when is_mutex_type t -> `Top
    | TInt (ik,_) -> `Int (ID.(cast_to ik (top ())))
    | TPtr _ -> `Address (if get_bool "exp.uninit-ptr-safe" then AD.(join null_ptr safe_ptr) else AD.top_ptr)
~~~
That, and arrays contain `bot` instead of `top`.

Reply:
I'm refactoring it.